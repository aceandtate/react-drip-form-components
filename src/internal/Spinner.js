import styled, { keyframes } from 'styled-components';
import { themeProps, bind } from '../theme';

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.div`
  display: block;
  position: absolute;
  top: 0.8em;
  right: 0.5em;
  margin: 0;
  width: 1em;
  height: 1em;
  border: 1px solid ${bind('spinnerColor')};
  border-top-width: 0;
  border-radius: 50%;
  animation: ${spin} ${bind('spinnerDuration')} infinite linear;
  transform-origin: 50% 49%;
`;

Spinner.defaultProps = themeProps;

export default Spinner;
