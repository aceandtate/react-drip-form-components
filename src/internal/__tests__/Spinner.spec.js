import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Spinner from '../Spinner';
import { mountWithTheme } from '../../__tests__/utils';


describe('Internal#<Spinner />', () => {
  test('Should be styled component', () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper).toMatchSnapshot();
  });


  test('Should be changed styled by theme', () => {
    const wrapper = mountWithTheme(<Spinner />, { spinnerColor: 'red' });
    expect(wrapper).toHaveStyleRule('border', '1px solid red');
  });
});
