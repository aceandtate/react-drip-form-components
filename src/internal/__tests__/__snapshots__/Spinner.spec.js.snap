// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal#<Spinner /> Should be styled component 1`] = `
.c0 {
  display: block;
  position: absolute;
  top: 0.8em;
  right: 0.8em;
  margin: 0;
  width: 1em;
  height: 1em;
  border: 1px solid rgb(0,188,212);
  border-top-width: 0;
  border-radius: 50%;
  -webkit-animation: iVXCSc 500ms infinite linear;
  animation: iVXCSc 500ms infinite linear;
  -webkit-transform-origin: 50% 49%;
  -ms-transform-origin: 50% 49%;
  transform-origin: 50% 49%;
}

<div
  className="c0"
/>
`;
